### DISCUSSION TITLE
WebGL application not working on github page

### DISCUSSION THREAD

**Discussion by SerenaXiu** [community/community#22540] [Other Feature Feedback, Questions, & Ideas]
<p>when I try to access the html located at “<a href="https://serenaxiu.github.io/Embedded-Tic-Tac-Toe/WebGL/EmWiApp.html" rel="noopener nofollow ugc">https://serenaxiu.github.io/Embedded-Tic-Tac-Toe/WebGL/EmWiApp.html</a>”, all what I get is a blank page. The WebGL application is not displayed at all</p>
<p>I opened the WebGL through the “live server” extension of Visual Studio Code, and the application is displayed and works as intended:<br>
<img src="https://user-images.githubusercontent.com/48486613/181071707-43eed490-55b3-4f27-8433-d4d8f6ea2065.png" alt="image" data-base62-sha1="iEwsMHyHeIHKqCwyhUZjfSLJ27D" width="465" height="416"></p>
<p>kind regards,<br>
SerenaXiu</p>

**Comment by jfullstackdev** [comment]
<p>you have several load errors, revealed by Developer Tools at the browser,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/2/62e987e3533af08e361ed21fb1fe50a0acb2de33.png" data-download-href="/uploads/short-url/e71822i2XI8iiHldefEyVGGvxoT.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181071708-a975130b-b68f-4ce0-9787-b4952f3489bc.png" alt="image" data-base62-sha1="e71822i2XI8iiHldefEyVGGvxoT" width="564" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/2/62e987e3533af08e361ed21fb1fe50a0acb2de33_2_564x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/6/2/62e987e3533af08e361ed21fb1fe50a0acb2de33_2_846x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/6/2/62e987e3533af08e361ed21fb1fe50a0acb2de33_2_1128x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/2/62e987e3533af08e361ed21fb1fe50a0acb2de33_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1175×1040 69.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>as I was looking at your repo, there are so many branches, which you might be confused which one is being hosted,</p>
<p>if you want to host your game at GitHub Pages, at the root, make sure that all URLs are correct, including if you have a folder structure,</p>
<p>I also have games hosted on GitHub Pages, for example,</p>
<p><a href="https://github.com/xdvrx1/breakout-game">https://github.com/xdvrx1/breakout-game</a></p>
<p>look at the repo structure,</p>
<p><img src="https://user-images.githubusercontent.com/47092464/181071711-058c1be2-5971-4e86-8401-4dd8972c9757.png" alt="image" data-base62-sha1="cBbkHBHMyttt9oscDNIcOFZOF0u" width="613" height="267"></p>

**Comment by SerenaXiu** [comment]
<p>Hi!<br>
I didn’t give much detail about it, so I selected the branch webGL to be presented as the branch to be shown in github page</p>
<p>It really was time to clean up my repo a bit, so I did as you suggested. So removing one possiblity for errors to occur.</p>
<p>Through that I discovered the reason for the error msg, it seems to come from<br>
“EmWinApp” not being initiated.<br>
One of the error msgs is from this line of code (included in the file GameFontSmall.js):</p>
<pre><code class="lang-auto">var EmWiApp;if(!EmWiApp)throw new Error("The application file '_project.js' isn't yet loaded!"
);if(EmWiApp.uT)throw new Error("The file file 'GameFontSmall.js' included twice!"
);
</code></pre>
<p>In the EmWiApp.html the EmWinApp should have been initiated:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/6/06c436523dd8a15c149ca44fa76606cb09c8710d.png" data-download-href="/uploads/short-url/XRflpDV1BpESdOqCzqDxTFbcMB.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/48486613/181071715-26b2fcd3-49a9-4acd-87bc-761b4ea24e63.png" alt="image" data-base62-sha1="XRflpDV1BpESdOqCzqDxTFbcMB" width="690" height="478" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/6/06c436523dd8a15c149ca44fa76606cb09c8710d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">750×520 38.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>Do you maybe know the reason, why this was not initiated?</p>
<p>kind regards,<br>
SerenaXiu</p>

**Comment by jfullstackdev** [comment, marked as answer]
<p>all of them are loading except <code>_project.js</code> because you need a special <code>.nojekyll</code> to read it correctly,</p>
<blockquote>
<p>It is now possible to completely bypass Jekyll processing on <a href="http://pages.github.com/">GitHub Pages</a> by creating a file named <code>.nojekyll</code> in the root of your pages repo and pushing it to GitHub. This should only be necessary if your site uses files or directories that start with underscores since Jekyll considers these to be special resources and does not copy them to the final site.</p>
</blockquote>
<p><a href="https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/">https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/</a></p>
<p>put that at the root of webGL branch, because you are hosting it there not in main,</p>

**Comment by SerenaXiu** [comment]
<p>thanks a lot, this solved the problem!</p>

**Comment by jfullstackdev** [comment]
<p>also, I noticed you have the README as your landing page, since you added <code>.nojekyll</code>, you will be needing to create a simple <code>index.html</code> so that you still have your landing page on GitHub Pages, because the README is not being rendered anymore,</p>
<p>but the README is still the default when viewed as a repository, so if you don’t want to create the <code>index.html</code>, simply provide direct link of the game on the README,</p>
<p><a href="https://serenaxiu.github.io/Embedded-Tic-Tac-Toe/EmWiApp.html">https://serenaxiu.github.io/Embedded-Tic-Tac-Toe/EmWiApp.html</a></p>
<p>the current link is the link to the file, not to the live site where your game is running,</p>
<p>also, you next problem is that, you have two branches, which might confuse visitors where your game is located,</p>
